dsn.go

```diff
@@ -60,6 +60,8 @@ type Config struct {
 	MultiStatements         bool // Allow multiple statements in one query
 	ParseTime               bool // Parse time values to time.Time
 	RejectReadOnly          bool // Reject read-only connections
+
+	DialogFunc func(byte, string) (string, error) // Optional dialog auth implementation
```

Comment:
This should document what the default behavior is if no dialog auth implementation is provided